### PR TITLE
Disable recycling, refactoring & fixes

### PR DESCRIPTION
**Pull request by hshristov**
- Added tests for native view recycling
 - Disabled android native view recycling 
 - Move toString from view-common to view-base
 - Fix crash on application restore and navigation back on API26
 - Added setAsRootView method 
 - Added missing logo into perf-tests/recycling app
- additional fix for image-source-tests. ios is case sensitive.

### CODE REVIEW

**Comment by hshristov** [comment]
uitests

File: tns-core-modules/ui/core/view-base/view-base.d.ts [hunk 1]
```diff
@@ -306,6 +306,7 @@ export abstract class ViewBase extends Observable {
     public ensureDomNode();
 
     //@private
+    public _isPaddingRelative: boolean;
```
Comment (PanayotCankov) [inline tns-core-modules/ui/core/view-base/view-base.d.ts]:
```
/**
 * @private
 */
```

File: tns-core-modules/ui/core/view/view.android.ts [hunk 2]
```diff
@@ -486,7 +486,11 @@ export class View extends ViewCommon {
         } else {
             const nativeView = this.nativeViewProtected;
             org.nativescript.widgets.ViewHelper.setBackground(nativeView, value);
-            nativeView.setPadding(this._defaultPaddingLeft, this._defaultPaddingTop, this._defaultPaddingRight, this._defaultPaddingBottom);
+             if (this._isPaddingRelative) {
```
Comment (PanayotCankov) [inline tns-core-modules/ui/core/view/view.android.ts]: this.padding?

**Comment by hshristov** [comment]
uitests

**Pull request merged**